lib.php

```diff
@@ -96,63 +97,66 @@ function accredible_update_instance($post) {
+                    // Create the credential.
+                    $result = create_credential($user, $groupid, null, $completeddate);
+                    $credentialid = $result->id;
                     // Evidence item posts.
                     if ($post->finalquiz) {
                         $quiz = $DB->get_record('quiz', array('id' => $post->finalquiz), '*', MUST_EXIST);
-                        $users_grade = min( ( quiz_get_best_grade($quiz, $user->id) / $quiz->grade ) * 100, 100);
-                        $grade_evidence = array('string_object' => (string) $users_grade, 'description' => $quiz->name, 'custom' => true, 'category' => 'grade');
-                        if ($users_grade < 50) {
-                            $grade_evidence['hidden'] = true;
+                        $usersgrade = min( ( quiz_get_best_grade($quiz, $user->id) / $quiz->grade ) * 100, 100);
+                        $gradeevidence = array('string_object' => (string) $usersgrade,
+                            'description' => $quiz->name, 'custom' => true, 'category' => 'grade');
+                        if ($usersgrade < 50) {
+                            $gradeevidence['hidden'] = true;
                         }
-                        accredible_post_evidence($credential_id, $grade_evidence, true);
+                        accredible_post_evidence($credentialid, $gradeevidence, true);
                     }
-                    if ($transcript = accredible_get_transcript($post->course, $user_id, $post->finalquiz)) {
-                        accredible_post_evidence($credential_id, $transcript, true);
+                    if ($transcript = accredible_get_transcript($post->course, $userid, $post->finalquiz)) {
+                        accredible_post_evidence($credentialid, $transcript, true);
                     }
-                    accredible_post_essay_answers($user_id, $post->course, $credential_id);
-                    accredible_course_duration_evidence($user_id, $post->course, $credential_id, $completed_timestamp);
-                } else if ($accredible_certificate->achievementid) {
+                    accredible_post_essay_answers($userid, $post->course, $credentialid);
+                    accredible_course_duration_evidence($userid, $post->course, $credentialid, $completedtimestamp);
+                } else if ($accrediblecertificate->achievementid) {
                     if ($post->finalquiz) {
                         $quiz = $DB->get_record('quiz', array('id' => $post->finalquiz), '*', MUST_EXIST);
                         $grade = min( ( quiz_get_best_grade($quiz, $user->id) / $quiz->grade ) * 100, 100);
-                    } else {
+                    } // else?
                         // This is an older activity that now uses the course completion to issue cert.
                         // Can't use accredible_issue_default_certificate, but create_credential only works with groupid.
```

Comment:
@naveenbalachandar - In my local environment, `// else?` causes two warnings as shown below. I feel like these comments don't give useful information to us. Maybe we can remove the comments (Line#147-149) since we need to understand the whole logic to add features anyway. What do you think?

![Screen Shot 2021-09-23 at 14 20 25](https://user-images.githubusercontent.com/20009662/134506707-897f6088-cec5-4910-8255-1131fce2ae7f.png)

Reply:
For some reason this error didn't show in my local env. I have made the changes.